packages/core/bootstrap/src/lib/rate-limit/index.ts

```diff
@@ -95,12 +97,7 @@ export const withRateLimit = (store: Store<RootState>): Middleware => async (exe
   state = store.getState()
   logRemainingCapacity(state.heartbeats, IntervalNames.MINUTE)
 
-  const isCacheHit = !!result.maxAge
-  if (!isCacheHit) {
-    metrics.rateLimitCreditsSpentTotal
-      .labels({ id: input.id, participantId: requestTypeId, experimental: 'true' })
-      .inc(result.data.cost || 1)
-  }
+  metrics.observeMetrics(input.id, requestTypeId, result)
```

Comment:
I much rather prefer the style from the left, than using a helper function like this, as it gives a lot more information about what is actually happening here. Makes the code more readable and understandable. Just my 2c.